Inspicio/Controllers/ImagesController.cs

```diff
@@ -249,132 +282,63 @@ public class DataFromBody
             }
 
-            // if the review has been rejected
-            if (data.ThumbChosen == (int)Status.REJECTED)
+            if ( data.state == States.NeedsWork )
             {
-                if (review.Rejected == false)
-                {
-                    if (review.NeedsWork == true)
-                    {
-                        review.Rejected = true;
-                        review.NeedsWork = false;
-                        image.NoOfRejections++;
-                        if (image.NoOfNeedsWork > 0)
-                        {
-                            image.NoOfNeedsWork--;
-                        }
-                    }
-                    else if (review.Approved == true)
-                    {
-                        review.Approved = false;
-                        review.Rejected = true;
-                        image.NoOfRejections++;
-                        if (image.NoOfApprovals > 0)
-                        {
-                            image.NoOfApprovals--;
-                        }
-                    }
-                    else if ((review.Approved == false) && (review.NeedsWork == false) && (review.Rejected == false))
-                    {
-                        review.Rejected = true;
-                        image.NoOfRejections++;
-                    }
-                }
+                 if (review.State != Review.States.NeedsWork)
+                 {
+                    review.State = Review.States.NeedsWork;
+                 }
             }
 
-            // if the review needs more work
-            if (data.ThumbChosen == (int)Status.NEEDSWORK)
+            if (data.state == States.Rejected)
             {
-                if (review.NeedsWork == false)
+
+                if (review.State != Review.States.Rejected)
                 {
-                    if (review.Approved == true)
-                    {
-                        review.NeedsWork = true;
-                        review.Approved = false;
-                        image.NoOfNeedsWork++;
-                        if (image.NoOfApprovals > 0)
-                        {
-                            image.NoOfApprovals--;
-                        }
-                    }
-                    else if (review.Rejected == true)
-                    {
-                        review.Rejected = false;
-                        review.NeedsWork = true;
-                        image.NoOfNeedsWork++;
-                        if (image.NoOfRejections > 0)
-                        {
-                            image.NoOfRejections--;
-                        }
-                    }
-                    else if ((review.Approved == false) && (review.NeedsWork == false) && (review.Rejected == false))
-                    {
-                        review.NeedsWork = true;
-                        image.NoOfNeedsWork++;
-                    }
+                    review.State = Review.States.Rejected;
                 }
```

Comment:
Unfortunately js doesn't seem to like enums that much, so the last option doesn't seem possible. However, the second one is working so thanks! I'll upload the new version soon.